File: Python/Easy/ASCII_Art.py
import sys
import math


def getPos(mark):
    a = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ?'
    m = mark.upper()
    if m in a:
        return int(a.index(m))
    else:
        return int(len(a)-1)

l = int(input())
h = int(input())
t = input()
for i in range(h):
    row = input()
    line =''
    for ch in t:
        s = getPos(ch)*l
        end = s+l
        line+=row[s:end]
    print(line)

File: Java/Easy/ASCII_Art.java
import java.util.*;
class Solution {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int L = in.nextInt();
        int H = in.nextInt();
        if (in.hasNextLine()) {
            in.nextLine();
        }
        String T = in.nextLine();
        for (int i = 0; i < H; i++) {
            String line = "";
            String ROW = in.nextLine();
            for(char c : T.toCharArray()){
                int start = getPos(c)*L;
                int end = (start+L);
                line+=ROW.substring(start,end);
            }
            System.out.println(line);
        }
    }
    
        public static int getPos(char mark){
        String a= "ABCDEFGHIJKLMNOPQRSTUVWXYZ?";
        char m = Character.toUpperCase(mark);
        int r = 69;
        for(int x=0;x<a.length();x++){
            if(a.charAt(x)==m){
                r = x;
            }
        }
        if(r==69)
            return a.length()-1;
        else
            return r;
    }
}